src/commands/pull.js

```diff
@@ -75,7 +77,14 @@ class PullCommand extends ApifyCommand {
 
                     if (!file.folder) {
                         const fileContent = file.format === 'BASE64' ? Buffer.from(file.content, 'base64') : file.content;
-                        fs.writeFileSync(`${dirpath}/${file.name}`, fileContent);
+
+                        if (file.name === LOCAL_CONFIG_PATH) {
+                            const actorJson = JSON.parse(fileContent);
```

Comment:
Same issue as in https://github.com/apify/apify-core/pull/11001, `actor.json` can be JSON5, so this would fail in that case.